### ISSUE TITLE
Editor crashes when zooming in with block toolbar visible

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#22258]
**Describe the bug**

When a block toolbar is visible, zooming in to the point where the toolbar moves to the top of the page causes the editor to crash. The following error displays in the console: 

```
Uncaught Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
```

```
The above error occurred in the <Slot> component:
    in Slot (created by Slot)
    in Slot (created by BlockControlsSlot)
    in BlockControlsSlot (created by BlockControlsSlot)
    in BlockControlsSlot (created by BlockToolbar)
    in div (created by BlockToolbar)
    in BlockToolbar (created by HeaderToolbar)
    in div (created by HeaderToolbar)
    in div (created by NavigableContainer)
    in NavigableContainer (created by ForwardRef(NavigableContainer))
    in ForwardRef(NavigableContainer) (created by ForwardRef(NavigableMenu))
    in ForwardRef(NavigableMenu) (created by NavigableToolbar)
    in NavigableToolbar (created by HeaderToolbar)
    in HeaderToolbar (created by Header)
    in div (created by Header)
    in div (created by Header)
    in Header (created by Layout)
    in div (created by InterfaceSkeleton)
    in div (created by InterfaceSkeleton)
    in InterfaceSkeleton (created by NavigateRegions(InterfaceSkeleton))
    in div (created by NavigateRegions(InterfaceSkeleton))
    in NavigateRegions(InterfaceSkeleton) (created by Layout)
    in div (created by FocusReturnProvider)
    in FocusReturnProvider (created by Layout)
    in Layout (created by Editor)
    in ErrorBoundary (created by Editor)
    in BlockEditorProvider (created by WithDispatch(BlockEditorProvider))
    in WithDispatch(BlockEditorProvider)
    in Unknown (created by Context.Consumer)
    in WithRegistryProvider(WithDispatch(BlockEditorProvider)) (created by EditorProvider)
    in BlockContextProvider (created by EditorProvider)
    in EntityProvider (created by EditorProvider)
    in EntityProvider (created by EditorProvider)
    in EditorProvider (created by WithDispatch(EditorProvider))
    in WithDispatch(EditorProvider)
    in Unknown (created by WithSelect(WithDispatch(EditorProvider)))
    in WithSelect(WithDispatch(EditorProvider))
    in Unknown (created by Context.Consumer)
    in WithRegistryProvider(WithSelect(WithDispatch(EditorProvider))) (created by Editor)
    in div (created by DropZoneProvider)
    in DropZoneProvider (created by Editor)
    in SlotFillProvider (created by SlotFillProvider)
    in SlotFillProvider (created by Editor)
    in StrictMode (created by Editor)
    in Editor (created by WithDispatch(Editor))
    in WithDispatch(Editor)
    in Unknown (created by WithSelect(WithDispatch(Editor)))
    in WithSelect(WithDispatch(Editor))
```

**To reproduce**
1. Check that "Top toolbar" is NOT selected in editor options
2. Add a block on the page and interact with it so that its toolbar appears
3. With the block toolbar visible, start zooming in until editor crashes. This may take more or less zooming depending on window width.

**Expected behavior**

Expected editor to not crash when zooming in.

**Screenshots**

**Editor version (please complete the following information):**

Latest Gutenberg plugin.

**Desktop (please complete the following information):**

Any desktop browser/version.

**Comment by kirilzh** [comment]
Looks like it's been fixed with the release of 8.1.0. Could not reproduce after upgrading.

**Comment by tellthemachines** [comment]
Hmm, I can still reproduce this both on a regular WP install with 8.1.0 and on my local dev environment with the up-to-date master branch.

**Comment by diegohaz** [comment]
Seems related to this line: https://github.com/WordPress/gutenberg/blob/01d2de2c4a752a426fbbf7a3656a37b274ed9d3b/packages/components/src/slot-fill/bubbles-virtually/slot.js#L38
